src/teaching-element/Assessment/MatchingQuestion.vue

```diff
@@ -124,16 +125,29 @@ export default {
       let { premises, responses } = this;
       this.source = shuffle(premises.map(it => ({ ...it, dragged: false })));
       this.target = sortBy(responses.map(it => ({ ...it, answers: [] })), 'key');
+    },
+    initializeSubmission(val) {
+      if (!val) return;
+      Object.keys(val).forEach(key => {
+        const source = this.source.find(it => it.key === key);
+        const target = this.target.find(it => it.key === val[key]);
+        target.answers.push(source);
+        source.dragged = true;
+      });
     }
   },
   created() {
     this.initialize();
+    this.$nextTick(() => this.initializeSubmission(this.submission));
```

Comment:
Why the `$nextTick` here? If it's not needed you can use the approach from [above]().

Reply:
`nextTick` is to wait for the initialize to finish setting up `source` and `target` arrays so we can modify those arrays.

Reply:
`nextTick` is used when you want to wait on DOM to update. Check [the docs](https://vuejs.org/v2/api/#vm-nextTick). Since you don't rely here on DOM, but rather on the dataset state alone you don't need it.